overlord/snapstate/snapstate.go

```diff
@@ -1723,11 +1723,17 @@ func Remove(st *state.State, name string, revision snap.Revision) (*state.TaskSe
 	}
 
 	if tp, _ := snapst.Type(); tp == snap.TypeApp && removeAll {
-		ts, err := AutomaticSnapshot(st, name)
+		expiration, err := AutomaticSnapshotExpiration(st)
 		if err != nil {
 			return nil, err
 		}
-		addNext(ts)
+		if expiration > 0 {
+			ts, err := AutomaticSnapshot(st, name)
```

Comment:
this is a bit of weird layering, it's probably simpler if AutomaticSnapshot returns instead a new snapstate.ErrNothingToDo